### ISSUE TITLE
[dev.icinga.com #7860] notification.vars are not accessible from within a NotificationCommand

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#2339]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/7860

**Created by agherardi on 2014-11-26 17:01:31 +00:00**

Assignee: _agherardi_
Status: _Resolved_  (closed on _2014-12-02 10:07:35 +00:00_)
Target Version: _(none)_
Last Update: _2014-12-02 10:07:35 +00:00 (in Redmine)_

    Icinga Version: 2.1.0-1

---

I'm trying to figure out if it's possible to pass custom parameters from a Notification to a NotificationCommand. I tried the following:

    object NotificationCommand "my_notify" {
      import "plugin-notification-command"

      command = [ PluginDir + "/my_notify" ]

      arguments = {
        "-type" = "$notification.vars.mytype$"
        "-source" = "$address$"
        "-time" = "$icinga.long_date_time$"
        "-properties" = "$notification.vars.properties$"
      }
    }

    apply Notification "notify_my_service_down" to Service {
      command = "my_notify"
      vars.mytype = "my_service_down"
      vars.properties = ""
      period = "24x7"
      interval = 0 // disable re-notification
      states = [ Critical ]
      types = [ Problem ]
      users = [ "my_user" ]
      assign where match(service.name, "My Service")
    }

icinga2.log shows that it ran:

    '/usr/lib/nagios/plugins/my_notify' '-source' '192.168.240.24' '-time' '2014-11-25 12:51:52 -0800' '-type' ''

So the type is an empty string instead of the expected my\_service\_down.

my\_notify takes the following parameters:
-source IP address of the host
-time timestamp
-type a string that represents our own, non-Icinga notification type
-properties some optional properties

Basically, I'm trying to map Icinga notifications - based on notification state, type and service that triggered it - to our own internal notifications. If possible, I'd like to define that mapping via the Icinga configuration.
